core/vm/evm_test.go

```diff
@@ -21,3 +29,27 @@ func TestAffectedMode_Update_whenTypical(t *testing.T) {
 		}
 	}
 }
+
+func TestCall_shouldReturnErrorNotAuthorized_whenNoCodeAndSupportsMultitenancy(t *testing.T) {
```

Comment:
Minor: I don't remember seeing this naming convention for test names in Go. Have you based on something standard?

Reply:
checking the naming convention on test names for Go, I see that there is no specific one apart from calling `TestMethod` and that is generic enough to create inconsistencies. I followed a standard approach like `Should_ExpectedBehavior_When_StateUnderTest` but then applied similar to what other tests look like in Go. I have used similar to this (sometimes with or without underscore or different naming but same goal) across other languages (except JS where when using `jest` you can put a string there with BDD style).

what other naming convention I should use to explain in a BDD style what this test is doing without having to read the code? or is it a matter only about the underscores?

thanks 👍

Reply:
for that standard, is this one I have used for a long time: https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html. in Go `TestUnitOfWork_StateUnderTest_ExpectedBehavior`

Reply:
Thanks.

I am good to go this way.

Please keep in mind that my favorite approach for testing in Go so far has been table testing (c.f. https://github.com/golang/go/wiki/TableDrivenTests)

In which case you can test multiple behaviors failing and non-failing in a single test function

Not done this way in the link but you can add a `desc string` field to the table items, so you can log it in case the test on an item does not assert properly.